dashboard/app/controllers/lessons_controller.rb

```diff
@@ -17,7 +17,7 @@ def disallow_legacy_script_levels
   # GET /s/script-name/lessons/1
   def show
     script = Script.get_from_cache(params[:script_id])
-    return render :forbidden unless script.is_migrated
+    return render :forbidden unless script.is_migrated && script.access_allowed(current_user)
```

Comment:
instead of calling access_allowed, could this just be `can?(:read, script)`, kind of like here? https://github.com/code-dot-org/code-dot-org/blob/1c53fe220c877102931741ac8e3b5c54ffb75bce/dashboard/app/views/courses/show.html.haml#L19